File: parse_erqonomika.py

import json
import re

def parse_questions():
    input_file = r"c:\Users\KhayalTurkic\Desktop\imtahan-magistr\src\data\erqonomika-bilet.txt"
    output_file = r"c:\Users\KhayalTurkic\Desktop\imtahan-magistr\src\data\erqonomika-suallari.json"

    with open(input_file, 'r', encoding='utf-8') as f:
        content = f.read()

    questions = []
    # Identify blocks separated by multiple newlines
    # The format seems to vary slightly but generally it's numbered lines.
    # We can split by lines and look for "1.", "2." etc.
    
    lines = content.split('\n')
    current_question = None
    
    count = 1
    
    for line in lines:
        line = line.strip()
        if not line:
            continue
            
        # Check if line starts with a number followed by dot
        match = re.match(r'^(\d+)\.\s*(.*)', line)
        if match:
            question_text = match.group(2).strip()
            # Some questions might span multiple lines if they were wrapped, 
            # but looking at the file, they seem to be single lines mostly.
            
            # Extract category or just proceed. 
            # We don't have categories in the text, so we'll leave it generic or inferred?
            # Actually, let's just create the question object.
            
            # Clean up the question text
            if not question_text and len(line) > 3:
                 # Case where there might be no space after dot "2.Question"
                 question_text = line[line.find('.')+1:].strip()

            if question_text:
                questions.append({
                    "id": count,
                    "question": question_text,
                    "answer": "", # No answer provided in text
                    "category": "Erqonomika ümumi" # Placeholder category
                })
                count += 1
        else:
            # Maybe a continuation of previous question?
            if questions:
                questions[-1]["question"] += " " + line

    data = {
        "subject": "Erqonomika və texniki dizayn",
        "totalQuestions": len(questions),
        "questions": questions,
        "categories": ["Erqonomika ümumi"]
    }

    with open(output_file, 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)

    print(f"Parsed {len(questions)} questions to {output_file}")

if __name__ == "__main__":
    parse_questions()

File: parse_erqonomika.js

const fs = require('fs');
const path = require('path');

const inputFile = path.join(__dirname, 'src', 'data', 'erqonomika-bilet.txt');
const outputFile = path.join(__dirname, 'src', 'data', 'erqonomika-suallari.json');

try {
    const data = fs.readFileSync(inputFile, 'utf8');
    const lines = data.split('\n');
    const questions = [];
    let count = 1;

    for (let i = 0; i < lines.length; i++) {
        let line = lines[i].trim();
        if (!line) continue;

        // Regex to match lines starting with "1.", "2.", etc.
        const match = line.match(/^(\d+)\.\s*(.*)/);
        if (match) {
            let questionText = match[2].trim();
            // If the text is empty, check if it continues on the same line but regex missed it (unlikely with .* but safe to check)
            // Or if it continues closely.
            
            if (questionText) {
                questions.push({
                    id: count++,
                    question: questionText,
                    answer: "", // No answer provided in text
                    category: "Bilet Sualları", // Default category
                    keywords: []
                });
            }
        } else {
             // If line doesn't start with number, append to previous question if it looks like continuation
             if (questions.length > 0 && line.length > 1) {
                 // Check if it's not a page number or garbage
                 if (!line.match(/^\d+$/)) {
                    questions[questions.length - 1].question += " " + line;
                 }
             }
        }
    }

    const outputData = {
        subject: "Erqonomika və texniki dizayn",
        totalQuestions: questions.length,
        questions: questions,
        categories: ["Bilet Sualları"]
    };

    fs.writeFileSync(outputFile, JSON.stringify(outputData, null, 2), 'utf8');
    console.log(`Successfully parsed ${questions.length} questions to ${outputFile}`);

} catch (err) {
    console.error("Error processing file:", err);
}
